File: CommandWords.java
import java.util.HashMap;
/**
 * This class is part of the "FUGwACL Adventure" application. 
 * "FUGwACL Adventure" is a slightly less simple, text based adventure game.  
 * 
 * This class holds an enumeration of all command words known to the game.
 * It is used to recognise commands as they are typed in.
 *
 * @author  Michael Kölling, David J. Barnes, and Christian Byrne
 * @version 2023-12-03
 */

public class CommandWords
{
    // a HashMap the holds Stings of the command words

    private HashMap<String, CommandWord> validCommands;
    /**
     * Constructor - initialise the command words.
     */
    public CommandWords()
    {
        validCommands = new HashMap<>();
        validCommands.put("walk", CommandWord.GO);
        validCommands.put("help", CommandWord.HELP);
        validCommands.put("EndGame", CommandWord.QUIT);
        validCommands.put("back", CommandWord.BACK);
        validCommands.put("look", CommandWord.LOOK);
        validCommands.put("take", CommandWord.TAKE);
        validCommands.put("drop", CommandWord.DROP);
        validCommands.put("items", CommandWord.ITEMS);
        validCommands.put("use", CommandWord.USE);
        validCommands.put("talk", CommandWord.TALK);
        validCommands.put("sleep", CommandWord.SLEEP);

    }

    /**
     * Check whether a given String is a valid command word. 
     * @return true if a given string is a valid command,
     * false if it isn't.
     */
    public boolean isCommand(String aString)
    {
        return validCommands.containsKey(aString);
    }

    public CommandWord getCommandWord(String commandWord)
    {
        CommandWord command = validCommands.get(commandWord);
        if(command != null) {
            return command;
        }
        else {
            return CommandWord.UNKNOWN;
        }
    }

    public void showAll() 
    {
        for(String command : validCommands.keySet()) {
            System.out.print(command + "  ");
        }
        System.out.println();
    }

}

File: ConsoleApp_121_FinalProjectShell/CommandWords.cs
//Transcribed CommandWords.java file - Adam Abbadusky
namespace ConsoleApp_121_FinalProjectShell;

public class CommandWords 
{
    // a Dictionary that holds strings of the command words
    private readonly Dictionary<string, CommandWord> _validCommands;

    /// <summary>
    /// Constructor - initialise the command words.
    /// </summary>
    public CommandWords()
    {
        _validCommands = new Dictionary<string, CommandWord>
        {
            { "walk", CommandWord.GO }, //adds to dictionary with 'add' instead of 'put' - fixed, now uses collection initializer
            { "help", CommandWord.HELP },
            { "EndGame", CommandWord.QUIT }, //possible bug doesn't match "quit"
            { "back", CommandWord.BACK },
            { "look", CommandWord.LOOK },
            { "take", CommandWord.TAKE },
            { "drop", CommandWord.DROP },
            { "items", CommandWord.ITEMS },
            { "use", CommandWord.USE },
            { "talk", CommandWord.TALK },
            { "sleep", CommandWord.SLEEP }
        };
    }

    /// <summary>
    /// Check whether a given string is a valid command word.
    /// </summary>
    /// <returns>
    /// True if a given string is a valid command, false if it isn't.
    /// </returns>
    public bool IsCommand(string aString) //bug will throw an exception if aString is null, difference with Java hashmaps vs c# dictionary
    {
        return _validCommands.ContainsKey(aString);
    }


    /// <summary>
    /// Get the CommandWord associated with a given command string.
    /// </summary>
    public CommandWord GetCommandWord(string commandWord)
    {
        // Bug:  Dictionary does not allow null keys and this will throw an exception if commandWord is null. 
        return _validCommands.ContainsKey(commandWord) ? _validCommands[commandWord] : CommandWord.UNKNOWN;
    }

    /// <summary>
    /// Print all valid commands to the console.
    /// </summary>
    public void ShowAll()
    {
        foreach (string command in _validCommands.Keys)
        {
            Console.Write(command + "  ");
        }
        Console.WriteLine();
    }

}